frontend/simple/js/backend/hapi.js

```diff
@@ -73,29 +73,55 @@ export class HapiBackend extends Backend {
   subscriptions () {
     return Object.keys(store.state.contracts)
   }
-  async subscribe (contractId: string) {
+  async subscribeAndSyncIdentity (username: string, ephemeral: bool) {
+    let identity = await namespace.lookup(username)
+    await this.subscribeAndSync(identity, ephemeral)
+  }
+  async unsubscribeIdentity (username: string, ephemeral: bool) {
+    let identity = await namespace.lookup(username)
+    await this.unsubscribe(identity, ephemeral)
+  }
```

Comment:
Contracts are contracts, we do not differentiate between them, so these functions need to go.